src/estimagic/optimization/tranquilo/tranquilo.py

```diff
@@ -213,78 +241,104 @@ def _tranquilo(
             state=state,
         )
 
-        if state.index not in filtered_indices:
-            raise ValueError()
+        n_effective_points = count_points(filtered_xs, trustregion=state.trustregion)
+
+        # ==============================================================================
+        # sample new points
+        # ==============================================================================
+        n_to_sample = max(0, target_sample_size - n_effective_points)
 
         new_xs = sample_points(
-            trustregion=trustregion,
-            target_size=target_sample_size,
+            trustregion=state.trustregion,
+            n_points=n_to_sample,
             existing_xs=filtered_xs,
             rng=sampling_rng,
         )
 
-        new_fvecs = [criterion(_x) for _x in new_xs]
-        new_indices = np.arange(history.get_n_fun(), history.get_n_fun() + len(new_xs))
-        history.add_entries(new_xs, new_fvecs)
+        # ==============================================================================
+        # criterion evaluations
+        # ==============================================================================
 
+        _, _, new_indices = wrapped_criterion(new_xs)
         model_indices = np.hstack([filtered_indices, new_indices])
-
         model_xs = history.get_xs(model_indices)
         model_fvecs = history.get_fvecs(model_indices)
 
-        centered_xs = (model_xs - trustregion.center) / trustregion.radius
+        # ==============================================================================
+        # build surrogate and optimize it
+        # ==============================================================================
+
+        weights = calculate_weights(model_xs, trustregion=state.trustregion)
+
+        centered_xs = (model_xs - state.trustregion.center) / state.trustregion.radius
```

Comment:
At some point, we should think about the name `centered_xs`. As we divide by the radius it is more closely related to a `standardized_xs` in my opinion. But it is not of major importance for now.

Reply:
Postponed.